src/v/cloud_topics/level_one/compaction/log_sampler.cc

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_topics/level_one/compaction/log_sampler.h"
+
+#include "cloud_topics/level_one/compaction/logger.h"
+#include "cloud_topics/level_one/compaction/meta.h"
+#include "compaction/utils.h"
+#include "config/configuration.h"
+#include "container/chunked_vector.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/timestamp.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+namespace cloud_topics::l1 {
+
+topic_cfg_provider_impl::topic_cfg_provider_impl(
+  cluster::metadata_cache* metadata_cache)
+  : _metadata_cache(metadata_cache) {}
+
+std::optional<std::reference_wrapper<const cluster::topic_configuration>>
+topic_cfg_provider_impl::get_topic_cfg(model::topic_namespace_view tp) const {
+    auto topic_md_ref = _metadata_cache->get_topic_metadata_ref(tp);
+    if (!topic_md_ref.has_value()) {
+        return std::nullopt;
+    }
+
+    return topic_md_ref.value().get().get_configuration();
+}
+
+log_sampler::log_sampler(
+  metastore* metastore,
+  std::unique_ptr<topic_cfg_provider> tp_metadata_provider)
+  : _metastore(metastore)
+  , _topic_metadata_provider(std::move(tp_metadata_provider)) {}
+
+ss::future<> log_sampler::sample_logs(
+  logs_type_t& logs,
```

Comment:
nit: not related to this PR but might be clearer if logs_type_t were log_set_t or something